distutils/dist.py

```diff
@@ -223,17 +223,17 @@ def __init__(self, attrs: MutableMapping[str, Any] | None = None) -> None:  # no
         # Distribution as a convenience to the developer.
         self.packages: list[str] | None = None
         self.package_data: dict[str, list[str]] = {}
-        self.package_dir: dict[str, str] | None = None
+        self.package_dir: Mapping[str, str] | None = None
         self.py_modules: list[str] | None = None
-        self.libraries = None
-        self.headers = None
+        self.libraries: list[tuple[str, dict[str, Any]]] | None = None
+        self.headers: list[str] | None = None
         self.ext_modules: list[Extension] | None = None
```

Comment:
typeshed typed this as `Sequence`, but as far as I can see
https://github.com/pypa/distutils/blob/603b94ec2e7876294345e5bceac276496b369641/distutils/command/build_ext.py#L178
and
https://github.com/pypa/distutils/blob/603b94ec2e7876294345e5bceac276496b369641/distutils/command/build_ext.py#L370-L383
means it *has* to be a list.